packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -90,10 +92,24 @@ final class VideoPlayer {
     }
 
     MediaSource mediaSource = buildMediaSource(uri, dataSourceFactory, formatHint, context);
+
     exoPlayer.setMediaSource(mediaSource);
     exoPlayer.prepare();
 
-    setupVideoPlayer(eventChannel, textureEntry);
+    setupVideoPlayer(exoPlayer, new QueuingEventSink());
+  }
+
+  VideoPlayer(
```

Comment:
I suppose this constructor could be used in non-test code, but it might be helpful to mark it as VisibleForTesting and/or add a comment explaining this constructor's purpose (why would I choose one constructor over the other?)